### ISSUE TITLE
c++ standards after c++17 not supported

### ISSUE THREAD

**Issue by madmongo1** [quicknir/wise_enum#40]
from wise_enum_detail.h

```
// optional type needed for interface
#ifndef WISE_ENUM_OPTIONAL_TYPE
#if __cplusplus == 201703L
#include <optional>
```

conditional should be >=

**Comment by samangh** [comment]
I think this issue has been fixed in commit 7fb8445644e54f77c23e6751fbc864d650b55db9 and can be closed.

**Comment by madmongo1** [comment]
Great, thanks.
